### PR TITLE
aur-sync: print dependencies by queue order

### LINKED ISSUE #414: Build order is inverted when installing poezio
Intended build order:
```sh
$ aur graph {poezio,python-aiodns,python-pycares,python-slixmpp}/.SRCINFO | tsort | tac
python-pycares
python-aiodns
python-slixmpp
poezio
```

but `aur sync poezio` starts by building poezio first. I am guessing the order is just reversed and @rafasc thinks a tac is probably missing. He bisected it to 5392072d287f6a0695f4ca3711faf70f75351010.

Note: currently `python-pycares` will fail to build due to a missing `makedepends` and the maintainer will mostly fix this soon.

### PR DESCRIPTION
**Pull request by AladW**
Fixes #414

### CODE REVIEW

File: lib/aur-sync [hunk 1]
```diff
@@ -58,15 +58,18 @@ db_fill_empty() {
...
-            if($1 in a && !($2 in seen)) {
-                print $0; seen[$2]
-            }
-        }' "$@"
+	map[$1] = $2
+	next
+    } {
+	# only print pkgbase on first occurence
+	if($1 in map && !(map[$1] in base)) {
+	    print $0
```
Comment (AladW) [inline lib/aur-sync]: This isn't quite right...

**Comment by AladW** [comment]
Should work now (and be slightly less unreadable)

**Comment by rafasc** [comment]
built `curr` and `poezio` successfully, which were the packages people mentioned on #aurutils.

**Comment by AladW** [comment]
Great, thanks! I should probably test with a more complex dependency chain (such as the usual ROS candidates) before releasing v2...

File: lib/aur-sync [hunk 2]
```diff
@@ -58,15 +58,24 @@ db_fill_empty() {
...
-        } {
-            if($1 in a && !($2 in seen)) {
-                print $0; seen[$2]
-            }
-        }' "$@"
+	map[$1] = $2
+	next
+    } {
+	if ($1 in map) {
+	    base = map[$1]
```
Comment (FichteFoll) [inline lib/aur-sync]: Ew, mixed indentation :stuck_out_tongue:

Comment (AladW) [reply lib/aur-sync]: Man, emacs ignores `dir-locals` when switching to `awk` mode for some reason :disappointed:

**Pull request merged**